### PR TITLE
Perf: Disable core kind registry

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

I noticed `corekinds` were eating a lot of heap when working on https://github.com/grafana/grafana/pull/78430. This reduces `go_memstats_heap_inuse_bytes` by about 40% by stopping initialization of the unused `corekinds` package. 

The left side is an idle grafana running current `main`, and the right is this branch:
image

We could also potentially save another 20% in the future with a `go.mod` replace of `apache/arrow` using [this commit](https://github.com/toddtreece/arrow/commit/08bed04bb9743b383898d2f51f6833f481b16592).

**Why do we need this feature?**

Performance optimization.

**Who is this feature for?**

Hopefully it makes life a little easier on some computers somewhere

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -466,7 +466,6 @@ func (hs *HTTPServer) registerRoutes() {
 			})
 
 			dashboardRoute.Post("/calculate-diff", authorize(ac.EvalPermission(dashboards.ActionDashboardsWrite)), routing.Wrap(hs.CalculateDashboardDiff))
-			dashboardRoute.Post("/validate", authorize(ac.EvalPermission(dashboards.ActionDashboardsWrite)), routing.Wrap(hs.ValidateDashboard))
```
Comment (ryantxu) [inline pkg/api/api.go:469]: +1 does not do anything useful now, and is not used.

File: pkg/services/dashboards/models.go [hunk 2]
```diff
@@ -260,15 +260,6 @@ type SaveDashboardCommand struct {
 	UpdatedAt time.Time
 }
 
-type ValidateDashboardCommand struct {
-	Dashboard string `json:"dashboard" binding:"Required"`
-}
-
-type TrimDashboardCommand struct {
```
Comment (ryantxu) [inline pkg/services/dashboards/models.go:267]: related orphan command

File: pkg/api/api.go (hunk 1 above)
Comment (toddtreece) [reply pkg/api/api.go:469]: nice, i wasn't sure about that one, but i was hoping it was the case.

File: pkg/cuectx/ctx.go [hunk 3]
```diff
@@ -27,21 +27,25 @@ var GoCoreKindParentPath = filepath.Join("pkg", "kinds")
...
-var rt *thema.Runtime
-var mu sync.Mutex
+var (
+	ctx  *cue.Context
+	rt   *thema.Runtime
+	once sync.Once
+)
+
+func initContext() {
+	once.Do(func() {
```
Comment (ryantxu) [inline pkg/cuectx/ctx.go:37]: ❤️

**Review by ryantxu — APPROVED** [review]

**Review by spinillos — APPROVED** [review]

**Pull request merged**